ehrapy/tools/supervised/_feature_importances.py

```diff
@@ -0,0 +1,157 @@
+            except ValueError as e:
+                raise ValueError(
+                    f"Feature {predicted_feature} is not continuous and conversion to float failed. Either change the prediction "
+                    f"type to 'categorical' or change the feature data type to a continuous type."
+                ) from e
+
+        if model == "regression":
+            predictor = LinearRegression(**kwargs)
+        elif model == "svm":
+            predictor = SVR(kernel="linear", **kwargs)
+        elif model == "rf":
+            predictor = RandomForestRegressor(**kwargs)
+
+    elif prediction_type == "categorical":
+        if not pd.api.types.is_categorical_dtype(data[predicted_feature].dtype):
+            try:
+                data[predicted_feature] = data[predicted_feature].astype("category")
+            except ValueError as e:
+                raise ValueError(
+                    f"Feature {predicted_feature} is not categorical and conversion to category failed. Either change the prediction "
+                    f"type to 'continuous' or change the feature data type to a categorical type."
+                ) from e
+
+        if data[predicted_feature].nunique() > 2 and model in ["regression", "svm"]:
+            raise ValueError(
+                f"Feature {predicted_feature} has more than two categories. Please choose random forest (rf) as model for multi-class classification."
+            )
+
+        if model == "regression":
+            predictor = LogisticRegression(**kwargs)
+        elif model == "svm":
+            predictor = SVC(kernel="linear", **kwargs)
+        elif model == "rf":
+            predictor = RandomForestClassifier(**kwargs)
+
+    if input_features == "all":
+        input_features = list(adata.var_names)
+        input_features.remove(predicted_feature)
+
+    input_data = data[input_features]
+    labels = data[predicted_feature]
+
+    for feature in input_data.columns:
+        try:
+            input_data[feature] = input_data[feature].astype(float)
+
+            if feature_scaling is not None:
+                scaler = StandardScaler() if feature_scaling == "standard" else MinMaxScaler()
+                input_data[feature] = scaler.fit_transform(input_data[[feature]])
+        except ValueError as e:
+            raise ValueError(
+                f"Feature {feature} is not numeric. Please encode non-numeric features before calculating "
+                f"feature importances or drop them from the input_features list."
+            ) from e
+
+    x_train, x_test, y_train, y_test = train_test_split(input_data, labels, test_size=test_split_size)
+
+    predictor.fit(x_train, y_train)
+
+    score = predictor.score(x_test, y_test)
+    evaluation_metric = "R2 score" if prediction_type == "continuous" else "accuracy"
+    logg.info(f"Training completed. The model achieved an {evaluation_metric} of {score:.2f} on the test set.")
```

Comment:
We could consider mentioning the size of the test set? Not sure whether it's all that important...

Reply:
I think it's a nice idea since it provides information on the reliability of the test score. I've added it.